Update backup-and-restore-tool.md (#3539)
pingcap/docs: br/backup-and-restore-tool.md

- Currently, TiDB does not support backing up and restoring partitioned tables.
- Currently, you can perform restoration only on new clusters.
- It is recommended that you execute multiple backup operations serially. Otherwise, different backup operations might interfere with each other.

> - BR supports operations only between clusters with the same [`new_collations_enabled_on_first_bootstrap` value](/character-set-and-collation.md#collation-support-framework) because BR only backs up KV data. If the cluster to back up and the cluster to restore use different collations, the data validation fails. Therefore, before restoring a cluster, make sure that the switch value from the query result of the `select VARIABLE_VALUE from mysql.tidb where VARIABLE_NAME='new_collation_enabled';` statement is consistent with that during the backup process.

Reviewer: ```suggestion
- BR supports operations only between clusters with the same [`new_collations_enabled_on_first_bootstrap`](/character-set-and-collation.md#collation-support-framework) value because BR only backs up KV data. If the cluster to be backed up and the cluster to be restored use different collations, the data validation fails. Therefore, before restoring a cluster, make sure that the switch value from the query result of the `select VARIABLE_VALUE from mysql.tidb where VARIABLE_NAME='new_collation_enabled';` statement is consistent with that during the backup process.
```

> - BR only supports operations between clusters with the same `new_collations_enabled_on_first_bootstrap` [switch value](/character-set-and-collation.md#Collation support framework) because BR only backs up KV data. If the backup cluster and the recovery cluster use different sorting rules, the data verification fails. Therefore, when restoring the cluster, you need to make sure that the switch value query result of the `select VARIABLE_VALUE from mysql.tidb where VARIABLE_NAME='new_collation_enabled';` statement is consistent with the query result of the backup. Then you can perform the restoration operation.

Reviewer: ```suggestion
- BR supports operations only between clusters with the same [`new_collations_enabled_on_first_bootstrap` value](/character-set-and-collation.md#collation-support-framework) because BR only backs up KV data. If the cluster to back up and the cluster to restore use different collations, the data validation fails. Therefore, before restoring a cluster, make sure that the switch value from the query result of the `select VARIABLE_VALUE from mysql.tidb where VARIABLE_NAME='new_collation_enabled';` statement is consistent with that during the backup process.
```

>     - For v3.1 clusters, TiDB does not support new collation, so it can be considered that the new collation is not enabled.

Reviewer: ```suggestion
    - For v3.1 clusters, the new collation framework is not supported, so you can see it as disabled.
```

>     - For v4.0 clusters, check whether new collation is enabled through `SELECT VARIABLE_VALUE FROM mysql.tidb WHERE VARIABLE_NAME='new_collation_enabled';`.

Reviewer: ```suggestion
    - For v4.0 clusters, check whether the new collation is enabled by executing `SELECT VARIABLE_VALUE FROM mysql.tidb WHERE VARIABLE_NAME='new_collation_enabled';`.
```

>     For example, if data is backed up from a v3.1 cluster and will be restored to a v4.0 cluster. The `new_collation_enabled` value of the v4.0 cluster is `true`, which means that the new collation is enabled in the cluster to be restored when this cluster is created. If you perform the restore in this situation, an error might occur.

Reviewer: ```suggestion
    For example, assume that data is backed up from a v3.1 cluster and will be restored to a v4.0 cluster. The `new_collation_enabled` value of the v4.0 cluster is `true`, which means that the new collation is enabled in the cluster to be restored when this cluster is created. If you perform the restore in this situation, an error might occur.
```

>     For example, the data is backed up in a v3.1 cluster. If you restore the data to a v4.0 cluster and the value of `new_collation_enabled` in the query restore cluster is `true`. This means that the  switch supported by the new collation is enabled when the restore cluster is created. Restoring data at this time may cause errors.

Reviewer: ```suggestion
    For example, if data is backed up from a v3.1 cluster and will be restored to a v4.0 cluster. The `new_collation_enabled` value of the v4.0 cluster is `true`, which means that the new collation is enabled in the cluster to be restored when this cluster is created. If you perform the restore in this situation, an error might occur.
```
